iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/GeneratorXml.java

```diff
@@ -50,7 +50,7 @@ protected void writeRootElementAttributes(Generator g, VoltageLevel vl, NetworkX
 
     @Override
     protected void writeSubElements(Generator g, VoltageLevel vl, NetworkXmlWriterContext context) throws XMLStreamException {
-        if (!Objects.equals(g, g.getRegulatingTerminal().getConnectable())) {
+        if (g.getRegulatingTerminal() != null && !Objects.equals(g, g.getRegulatingTerminal().getConnectable())) {
```

Comment:
The condition should be different depending on XIIDM version

Reply:
This should change, shouldn't it? The terminal if it is local should be written in IIDM >= 1.6 in order to make the difference with a null (undefined) terminal. If the regulating terminal is null, it should throw an exception if IIDM < 1.6

Reply:
Done.

Until version 1.6 only remote `regulatingTerminals `were written. This generates an ambiguity in the reading process. In disabled voltage regulating controls the regulatingTerminal could be null or local. I decided to consider it null.

Since version 1.6 local and remote `regulatingTerminals `are written to eliminate the previous ambiguity. This modification of the code provokes adjustments in several xml and xiidm resource files.